hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/OMFailoverProxyProvider.java

```diff
@@ -191,20 +192,19 @@ public Text getCurrentProxyDelegationToken() {
   private Text computeDelegationTokenService() {
     // For HA, this will return "," separated address of all OM's.
     StringBuilder rpcAddress = new StringBuilder();
-    int count = 0;
+
     for (Map.Entry<String, OMProxyInfo> omProxyInfoSet :
         omProxyInfos.entrySet()) {
-      count++;
-      rpcAddress =
-          rpcAddress.append(
-              omProxyInfoSet.getValue().getDelegationTokenService());
+      Text dtService = omProxyInfoSet.getValue().getDelegationTokenService();
 
-      if (omProxyInfos.size() != count) {
-        rpcAddress.append(",");
+      // dtService can be null when during client object creation when one of
+      // the OM configured address in unreachable.
+      if (dtService != null) {
+        rpcAddress.append(",").append(rpcAddress);
```

Comment:
As we discussed offline, can you add a TODO to fix the computed DTService when the dtService is temporarily avialable due to OM instance reboot, DNS issue?

Reply:
I have added TODO in OMProxyInfo.java. Based on @arp7 offline comments removed TODO and create a Jira to resolve this.
https://issues.apache.org/jira/browse/HDDS-3137

Reply:
I have an in-progress patch for HDDS-3233. I need to test out the scenario, and I see few in cases not resolving the address. Still looking into that part, will post the change as part of HDDS-3233.